resource/graph_node.go

```diff
@@ -321,6 +328,26 @@ func (w *GraphNode) setNeedsReconfigure(newConfig Config, mustReconfigure bool,
 	w.unresolvedDependencies = dependencies
 }
 
+// UpdatePendingRevision sets the next revision to be applied once the node is in a
+// [NodeStateReady] state.
+func (w *GraphNode) UpdatePendingRevision(revision string) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+
+	w.pendingRevision = revision
+}
+
+// UpdateRevision updates the node config revision if the node is in a [NodeStateReady]
+// state.
+func (w *GraphNode) UpdateRevision(revision string) {
+	w.mu.Lock()
+	defer w.mu.Unlock()
+	if w.state == NodeStateReady {
+		w.pendingRevision = revision
```

Comment:
From what I can tell we only `UpdateRevision` within `local_robot.go`; is there any way we could set a pending revision and then erase it with this line? I guess more generally, why are we setting `pendingRevision` in addition to `revision` here?

Reply:
hm, interesting corner case - i don't think that can happen right now since on any given reconfiguration the method we call to set a revision is mutually exclusive:

* `UpdateRevision` if the resource is unchanged
* `UpdatePendingRevision` if the resource if new or modified
* none of these if the resource is being removed

thinking ahead, we will need to be careful about this when we introduce an `Unhealthy` state, since we might have situations where a resources is unmodified by a new config but is `Unhealthy` (not `Ready`) when it is supposed to receive it's new revision. I haven't quite thought about what to in that situation, but it depends on how we end up implementing `Unhealthy` so punting on that for now.